### PR TITLE
fix(rest): Fix self link for user resource

### PR DESCRIPTION
**Pull request by greimela-si**
Use the correct endpoint `/users/byid/:id` to access a user by id.

Signed-off-by: Andreas Greimel <external.andreas.greimel@bosch-si.com>

### CODE REVIEW

**Comment by maierthomas** [comment]
Here is my test result. The self link for users is changed to .../users/byid/<id>
Unfortunately the user id (in my case user email like admin1@sw360.org) encodes the "@" character to `%2540`.

image

To get it work I need to manuallly chang the value the id/email
From `admin1%2540sw360.org` to `admin1@sw360.org`

**Comment by maierthomas** [comment]
Here is my test result. The self link for users is changed to .../users/byid/<id>
Unfortunately the user id (in my case user email like admin1@sw360.org) encodes the "@" character to `%2540`.

image

To get it work I need to manuallly chang the value the id/email
From `admin1%2540sw360.org` to `admin1@sw360.org`

**Comment by maierthomas** [comment]
Here is my test result. The self link for users is changed to .../users/byid/<id>
Unfortunately the user id (in my case user email like admin1@sw360.org) encodes the "@" character to `%2540`.

To get it work I need to manuallly chang the value the id/email
From `admin1%2540sw360.org` to `admin1@sw360.org`

**Comment by maierthomas** [comment]
Here is my test result. The self link for users is changed to .../users/byid/<id>
Unfortunately the user id (in my case user email like admin1@sw360.org) encodes the "@" character to `%2540`.

image

To get it work I need to manuallly chang the value the id/email
From `admin1%2540sw360.org` to `admin1@sw360.org`

**Comment by maierthomas** [comment]
Here is my test result. The self link for users is changed to .../users/byid/<id>
Unfortunately the user id (in my case user email like admin1@sw360.org) encodes the "@" character to `%2540`.

image

To get it work I need to manuallly chang the value the id/email
From `admin1%2540sw360.org` to `admin1@sw360.org`

**Comment by maierthomas** [comment]
Here is my test result. The self link for users is changed to .../users/byid/<id>
Unfortunately the user id (in my case user email like admin1@sw360.org) encodes the "@" character to `%2540`.

image

To get it work I need to manuallly chang the value the id/email
From `admin1%2540sw360.org` to `admin1@sw360.org`

**Comment by maierthomas** [comment]
Here is my test result. The self link for users is changed to .../users/byid/<id>
Unfortunately the user id (in my case user email like admin1@sw360.org) encodes the "@" character to `%2540`.

image

To get it work I need to manuallly chang the value the id/email
From `admin1%2540sw360.org` to `admin1@sw360.org`

**Comment by greimela-si** [comment]
We could use a different self links in case the user id is equal to the email address.
@maierthomas Do you think a condition such as 
```
String link = user.getId().equals(user.getEmail())
                ? "api/users/" + URLEncoder.encode(user.getEmail(), "UTF-8")
                : "api/users/byid/" + user.getId();
```
will work in your case?

**Comment by maierthomas** [comment]
why not using `api/users/byid/` and always encode `user.getId()` ?

**Comment by greimela-si** [comment]
I think you mean not to encode `user.getId()` since it currently is encoded?
That should solve the problem, I updated the PR!

**Comment by maierthomas** [comment]
Tested successfully:

**List users**
image

**Get user by selflink**
image

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/integration/UserTest.java [hunk 1]
```diff
@@ -54,20 +57,35 @@ public void before() {
...
+        ResponseEntity<String> response = sendRequest("http://localhost:" + port + "/api/users");
         assertThat(HttpStatus.OK, is(response.getStatusCode()));
 
-		TestHelper.checkResponse(response.getBody(), "users", 2);
-	}
+        TestHelper.checkResponse(response.getBody(), "users", 2);
+    }
+
+    @Test
+    public void should_get_single_user_with_correct_self_link() throws IOException {
```
Comment (maierthomas) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/integration/UserTest.java]: I would rather name this method `should_get_single_user_by_user_self_link`

Comment (greimela-si) [reply rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/integration/UserTest.java]: I updated the name to clarify the test intention.

**Comment by maierthomas** [comment]
pipe crashed regarding some cert issues.. I restarted the job, lets see.

**Pull request merged**